pkg/liquidity-mining/contracts/GaugeAdder.sol

```diff
@@ -0,0 +1,168 @@
+
+        return false;
+    }
+
+    // Admin Functions
+
+    // Functions for the "LiquidityMiningCommittee" and "veBAL" types are purposefully omitted as there is
+    // no reason for new gauges to be deployed for these types so there is no need to expose methods to add them.
+
+    /**
+     * @notice Adds a new gauge to the GaugeController for the "Ethereum" type.
+     */
+    function addEthereumGauge(address gauge) external override authenticate {
+        _addGauge(gauge, GaugeType.Ethereum);
+    }
+
+    /**
+     * @notice Adds a new gauge to the GaugeController for the "Polygon" type.
+     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
+     * It should not be called with the address of the gauge which is deployed on Polygon
+     */
+    function addPolygonGauge(address rootGauge) external override authenticate {
+        _addGauge(rootGauge, GaugeType.Polygon);
+    }
+
+    /**
+     * @notice Adds a new gauge to the GaugeController for the "Arbitrum" type.
+     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
+     * It should not be called with the address of the gauge which is deployed on Arbitrum
+     */
+    function addArbitrumGauge(address rootGauge) external override authenticate {
+        _addGauge(rootGauge, GaugeType.Arbitrum);
+    }
+
+    /**
+     * @notice Adds `factory` as an allowlisted factory contract for gauges with type `gaugeType`.
+     */
+    function addGaugeFactory(ILiquidityGaugeFactory factory, GaugeType gaugeType) external override authenticate {
+        // Casting is safe as n_gauge_types return value is >= 0.
+        require(uint256(gaugeType) < uint256(_gaugeController.n_gauge_types()), "Invalid gauge type");
+
+        // Sanity check that calling `isGaugeFromFactory` won't revert
+        require(!factory.isGaugeFromFactory(address(0)), "Invalid factory implementation");
+
+        EnumerableSet.AddressSet storage gaugeFactories = _gaugeFactoriesByType[gaugeType];
+        require(gaugeFactories.add(address(factory)), "Factory already added");
+
+        emit GaugeFactoryAdded(gaugeType, factory);
+    }
+
+    // Internal functions
+
+    function _canPerform(bytes32 actionId, address account) internal view override returns (bool) {
+        return getAuthorizer().canPerform(actionId, account, address(this));
+    }
+
+    /**
+     * @dev Adds `gauge` to the GaugeController with type `gaugeType` and an initial weight of zero
+     */
+    function _addGauge(address gauge, GaugeType gaugeType) private {
+        require(isGaugeFromValidFactory(gauge, gaugeType), "Invalid gauge");
+
+        // `_gaugeController` enforces that duplicate gauges may not be added so we do not need to check here.
+        _authorizerAdaptor.performAction(
+            address(_gaugeController),
+            abi.encodeWithSelector(IGaugeController.add_gauge.selector, gauge, gaugeType, 0)
```

Comment:
```suggestion
            abi.encodeWithSelector(IGaugeController.add_gauge.selector, gauge, gaugeType)
```

This is technically incorrect since we're using the version of `add_gauge` that lacks a weight argument. abi.encode does not check times, so the compiler doesn't complain. If the Controller was written in Solidity, this would work as extra calldata is ignored - I'm not sure about Vyper.